From 056029c6506dd1e7bd285943099e9fcc091cff8b Mon Sep 17 00:00:00 2001
From: Marco Trevisan (Trevino) <mail@3v1n0.net>
Date: Thu, 18 Sep 2025 17:30:17 +0200
Subject: [PATCH] screencast: Disconnect signal handlers on data free

We may still receive dbus requests while the dialog handle has been
free'd, so disconnect any signal bound to to it when free'ing it
---
 src/screencast.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/screencast.c b/src/screencast.c
index 832a455..be9354f 100644
--- a/src/screencast.c
+++ b/src/screencast.c
@@ -100,6 +100,8 @@ is_screen_cast_session (Session *session)
 static void
 screen_cast_dialog_handle_free (ScreenCastDialogHandle *dialog_handle)
 {
+  g_signal_handlers_disconnect_by_data (dialog_handle->request, dialog_handle);
+  g_signal_handlers_disconnect_by_data (dialog_handle->dialog, dialog_handle);
   g_clear_pointer (&dialog_handle->dialog, gtk_window_destroy);
   g_clear_object (&dialog_handle->external_parent);
   g_object_unref (dialog_handle->request);
--
GitLab
